### PR TITLE
seastar: lower the required yaml-cpp version to 0.5.1

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
this change should address:

```
-- Found rt: rt  
CMake Error at /usr/share/cmake3/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find yaml-cpp: Found unsuitable version "0.5.1", but required is
  at least "0.5.2" (found /usr/include)
Call Stack (most recent call first):
  /usr/share/cmake3/Modules/FindPackageHandleStandardArgs.cmake:376 (_FPHSA_FAILURE_MESSAGE)
  src/seastar/cmake/Findyaml-cpp.cmake:42 (find_package_handle_standard_args)
  src/CMakeLists.txt:398 (_find_package)
  src/seastar/CMakeLists.txt:144 (find_package)


-- Configuring incomplete, errors occurred!
```

see https://jenkins.ceph.com/job/ceph-pull-requests/2100/consoleFull#-86978783385e1414f-af06-4588-8fed-ab5891ddc58a

File: cmake/modules/Findfmt.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  file(STRINGS "${fmt_INCLUDE_DIR}/fmt/format.h" fmt_VERSION_LINE
+    REGEX "^#define[ \t]+FMT_VERSION[ \t]+[0-9]+$")
+  string(REGEX REPLACE "^#define[ \t]+FMT_VERSION[ \t]+([0-9]+)$"
+    "\\1" fmt_VERSION "${fmt_VERSION_LINE}")
+  foreach(ver "fmt_VERSION_PATCH" "fmt_VERSION_MINOR" "fmt_VERSION_MAJOR")
+    math(EXPR ${ver} "${fmt_VERSION} % 100")
+    math(EXPR fmt_VERSION "(${fmt_VERSION} - ${${ver}}) / 100")
+  endforeach()
+  set(fmt_VERSION_STRING
+      "${fmt_VERSION_MAJOR}.${fmt_VERSION_MINOR}.${fmt_VERSION_PATCH}")
```
Comment (cbodley) [inline cmake/modules/Findfmt.cmake]: is this version stuff used anywhere?

Comment (tchaikov) [reply cmake/modules/Findfmt.cmake]: it's used in the following `find_package_handle_standard_args()` call.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2018-07-27-1103/c4c905775903abed8a3009116ff8ffb95463a6f1/

**Pull request merged**